From 489e933c98f45ca793a995f747282d764c020b5e Mon Sep 17 00:00:00 2001
From: Larry Finger <Larry.Finger@lwfinger.net>
Date: Sun, 23 Mar 2014 23:08:51 -0500
Subject: [PATCH 458/458] rtl8192du: Fix cfg80211 warhing

Since the CFG80211 option was turned on, the following warning was issued:

WARNING: CPU: 1 PID: 7599 at net/wireless/sme.c:847 __cfg80211_disconnected+0x3a8/0x3d0 [cfg80211]()

Signed-off-by: Larry Finger <Larry.Finger@lwfinger.net>
---
 os_dep/ioctl_cfg80211.c | 2 +-
 1 files changed, 1 insertions(+), 1 deletion(-)

--- a/drivers/staging/rtl8192du/os_dep/ioctl_cfg80211.c
+++ b/drivers/staging/rtl8192du/os_dep/ioctl_cfg80211.c
@@ -4102,7 +4102,7 @@ int rtw_wdev_alloc(struct rtw_adapter *p
 	}
 	wdev->wiphy = wiphy;
 	wdev->netdev = pnetdev;
-	wdev->iftype = NL80211_IFTYPE_MONITOR; /*  for rtw_setopmode_cmd() in cfg80211_rtw_change_iface() */
+	wdev->iftype = NL80211_IFTYPE_STATION;
 	padapter->rtw_wdev = wdev;
 	pnetdev->ieee80211_ptr = wdev;
 
